tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -448,6 +448,15 @@ internal sealed class AsyncToSyncMethodTransformerTests {
 		Assert.AreEqual( @"[Blocking] void Bar() { Console.WriteLine($""Hello, {GetUsername()}! Welcome back to {m_baz}, it is {GetTime()} {m_ampm}.""); }", actual.Value.ToFullString() );
 	}
 
+	[Test]
+	public void TaskCanceledException() {
+		var actual = Transform( @"[GenerateSync] async Task BarAsync() { try { response = await GetBazAsync().ConfigureAwait( false ); } catch( TaskCanceledException exception ) { Console.WriteLine( ""Failed"" ); } }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( @"[Blocking] void Bar() { try { response = GetBaz(); } catch( OperationCanceledException exception ) { Console.WriteLine( ""Failed"" ); } }", actual.Value.ToFullString() );

```

Comment:
Hmmmm.... I'm not sure about this one? Like maybe we should just exclude `catch( TaskCanceledException )` guards entirely? I'm not sure `OperationCanceledException` is really an equivalent.

@omsmith wdyt?

Reply:
I based this off of the fact that TaskCanceledException derives from OperationCanceledException: https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcanceledexception?view=net-7.0

Reply:
Yeah, I would generally expect TaskCanceledExceptions to disappear in entirely sync flows.

Anywhere we're just doing `Task.Run( AsyncThing() ).Result` is still going to have them though.